### PR TITLE
REF: Remove BlockManager.rename_axis

### PR DESCRIPTION
**Pull request by jbrockmendel**
Better to do it using NDFrame methods

cc @toobaz I know you're on board for getting index/axis stuff out of BlockManager.

### CODE REVIEW

File: pandas/core/indexes/api.py [hunk 1]
```diff
@@ -311,3 +311,24 @@ def all_indexes_same(indexes):
         if not first.equals(index):
             return False
     return True
+
+
+def transform_index(index: Index, func, level=None) -> Index:
```
Comment (jreback) [inline pandas/core/indexes/api.py]: how is this different from .map?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/api.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/indexes/api.py]: Looks like the main difference is in the MultiIndex case passing level makes this operate on just that levle

Comment (jreback) [reply pandas/core/indexes/api.py]: k, it would be better to use map then than duplicating this code

Comment (jreback) [reply pandas/core/indexes/api.py]: or at least co-locate as a private function for future de-duplication

Comment (jbrockmendel) [reply pandas/core/indexes/api.py]: Updated

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
k thanks, with the followon comments about deduplicating (I see you made a comment)

**Pull request merged**